scaleway/data_source_image.go

```diff
@@ -59,19 +60,36 @@ func scalewayImageAttributes(d *schema.ResourceData, img *api.ScalewayImage) err
 func dataSourceScalewayImageRead(d *schema.ResourceData, meta interface{}) error {
 	scaleway := meta.(*Client).scaleway
 
-	var needle string
+	var nameMatch func(api.MarketImage) bool
 	if name, ok := d.GetOk("name"); ok {
-		needle = name.(string)
+		nameMatch = func(img api.MarketImage) bool {
+			return img.Name == name.(string)
+		}
 	} else if nameFilter, ok := d.GetOk("name_filter"); ok {
-		needle = nameFilter.(string)
+		exp := regexp.MustCompile(nameFilter.(string))
```

Comment:
It looks like this would panic if the user provided an invalid regexp for `name_filter`. Bad user input should not cause a panic, since a panic can cause data loss. Would prefer to use `regexp.Compile` and return a good error if it fails.